dlib/dnn/loss.h

```diff
@@ -1160,10 +1163,18 @@ namespace dlib
                             continue;
                         }
                         const size_t idx = (k*output_tensor.nr() + p.y())*output_tensor.nc() + p.x();
+                        if (unique_truth_idxs.find(idx) != unique_truth_idxs.end())
+                        {
+                            // Ignore duplicate truth box in feature coordinates.
+                            loss -= options.loss_per_missed_target;
+                            truth_idxs.push_back(0);
```

Comment:
Setting this to 0 is going to cause this to interact badly with the use of `truth_idxs` later (here https://github.com/davisking/dlib/pull/1896/files#diff-32a043d312608d62a296b05ce1bc096eR1239-R1244). That code later on is checking if truth boxes are being knocked out by NMS, this check here is essentially the same thing, so should probably be done in the same place. Effectively, we have a floor on how tight you can make the NMS because of the CNN's resolution.

So you could keep track of how many times one of these CNN output cells is hit by a truth box and if it's more than once pretend that the NMS triggered.  Then treat it the same way as the NMS check at https://github.com/davisking/dlib/pull/1896/files#diff-32a043d312608d62a296b05ce1bc096eR1239-R1244 does (with an updated warning message to say why the box is being ignored).

Reply:
Hmm those links don't appear to lead anywhere?

Reply:
But no worries, I think I get it anyway. Will push some suggestion.

Reply:
Hmm what if I just push `idx` and not `0` here?

Reply:
Huh, yeah, links worked at first.  Anyway, I mean down at line 1228 where `truth_idxs` is used.

Reply:
Tried to do a minimal (yet correct) change.